pilot/helpers/Project.py

```diff
@@ -447,15 +448,18 @@ def ask_for_human_intervention(self, message, description=None, cbs={}, convo=No
                     raise e
 
     def log(self, text, message_type):
-        if self.ipc_client_instance is None or self.ipc_client_instance.client is None:
-            print(text)
-        else:
+        if self.check_ipc():
             self.ipc_client_instance.send({
                 'type': MESSAGE_TYPE[message_type],
                 'content': str(text),
             })
             if message_type == MESSAGE_TYPE['user_input_request']:
                 return self.ipc_client_instance.listen()
+        else:
+            print(text)
+
+    def check_ipc(self):
+        return self.ipc_client_instance is not None and self.ipc_client_instance.client is not None
```

Comment:
Have in mind that before on all places in code we were checking
`self.ipc_client_instance is None or self.ipc_client_instance.client is None`
and now we are checking the opposite.